dbms/src/DataStreams/NativeBlockOutputStream.cpp

```diff
@@ -101,8 +101,10 @@ void NativeBlockOutputStream::write(const Block & block)
         /// Send data to old clients without low cardinality type.
         if (remove_low_cardinality || (client_revision && client_revision < DBMS_MIN_REVISION_WITH_LOW_CARDINALITY_TYPE))
         {
-            column.column = recursiveRemoveLowCardinality(column.column);
-            column.type = recursiveRemoveLowCardinality(column.type);
+            if (auto col = recursiveRemoveLowCardinality(column.column.get()))
```

Comment:
Don't understand the change.

Reply:
The main idea is that function `recursiveRemoveLowCardinality` has changed it's interface. Before function was `ColumnPtr -> ColumnPtr`, and result was a column without `LowCardinality`. Now function is `IColumn * -> ColumnPtr`, and it returns new column only if argument had LowCardinality, and `nullptr` otherwise.

Reasons:
* If column doesn't contain `LowCardinality`, do not create new column object (less impact on ordinary columns).
* Can use with raw ptr argument (which was used in Aggregator)